### PR TITLE
Highlight words on playback

### PR DESCRIPTION
**Pull request by rileykeane**
## Summary

Highlights the word in the transcription that is currently playing. 

Please note, I do intended to animate the transition between words, however wanted to get this feature done for the MVP. For now the switch from one word to another feels a little janky. I plan to improve this by having the highlight slide over to the next word. 

I also had to make a tiny change to the PASTE_WORD and UNDO_PASTE_WORD ops to ensure the output start word time was updated - otherwise it caused some super weird behavior with the highlighting. 

<hr/>

### Why is this change needed?

This is feature helps the editor easily identify the word that's currently playing and enables them to follow along the transcript as the video is playing. 

The feature was ✨ _inspired_ ✨by an unnamed competitor. 

<hr/>

### What did you change?

Added a use effect hook that's dependent on the playback time. This will find the word thats currently playing and store its index in nowPlayingWordIndex. This is passed to the transcription block which will set the background color of the now playing word to orange 500 with a 50% opacity (same as the on hover color)

Also modified PASTE_WORD and UNDO_PASTE_WORD ops to return the processed transcription to ensure the new output start time was updated in store. 

<hr/>

### Does it depend on any other changes/PRs?

Nope.

<hr/>

### Screenshots of UI changes, if any

https://user-images.githubusercontent.com/27614346/169824053-713757a6-62cd-4bbc-b71a-dab4700a7480.mov

<hr/>

### Any sections in particular you want reviewed/discussed

I had refactored this and moved the logic into the transcription block itself, however moved it back into the project page because having it in the transcription block means I need to pass `time` in as a prop. This would mean the transcription block would re-render every time the `time` value changes (which is a lot...) so I reverted the change. 

I'm not that experienced with react, so someone who is, please advise on the best approach.  

<hr/>

### OS

- [ ] Linux
- [x] MacOS
- [ ] Windows

### CODE REVIEW

**Comment by chloebrett** [comment]
Task linked: [CU-2buvgk5 Highlight words on playback](https://app.clickup.com/t/2buvgk5)

**Comment by rileykeane** [comment]
test

File: src/renderer/pages/Project.tsx [hunk 1]
```diff
@@ -155,10 +156,22 @@ const ProjectPage = () => {
     };
   });
 
-  // TODO: figure out return type
+  useEffect(() => {
```
Comment (ghost) [inline src/renderer/pages/Project.tsx:160]: I understand for MVP but I reckon this should definitely be optimised if possible in a later PR. Please add a 'TODO: <comment-here>' comment to look into optimising this later

File: src/renderer/store/transcription/reducer.ts [hunk 2]
```diff
@@ -61,10 +61,11 @@ const transcriptionReducer: Reducer<Transcription | null, Action<any>> = (
 
     // Have to check this if to get rid of linter error
```
Comment (ghost) [inline src/renderer/store/transcription/reducer.ts]: No need for comment

File: src/renderer/store/transcription/reducer.ts [hunk 3]
```diff
@@ -61,10 +61,11 @@ const transcriptionReducer: Reducer<Transcription | null, Action<any>> = (
...
     if (pasteyWords !== undefined && suffix !== undefined) {
```
Comment (ghost) [inline src/renderer/store/transcription/reducer.ts]: Please update this to 'pasted words' - is this your spelling mistake? 😄

Comment (rileykeane) [reply src/renderer/store/transcription/reducer.ts]: that is left over from @Aretle

Comment (ghost) [reply src/renderer/store/transcription/reducer.ts]: 🚌

**Review by ghost — APPROVED** [review]
Nice work. Let's get it merged ASAP

**Pull request merged**